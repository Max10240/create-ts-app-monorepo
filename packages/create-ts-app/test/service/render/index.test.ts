import { ETplName, RenderService } from '@/service/render/index.js';
import { expect, test } from 'vitest';
import * as process from 'process';

const renderService = new RenderService(process.cwd());

test('render tsconfig', async () => {
  const contentCjsES2017 = await renderService.render(ETplName.tsconfigJson, {
    module: 'commonjs',
    target: 'es2017',
  });
  const contentEsm = await renderService.render(ETplName.tsconfigJson, {
    module: 'esnext',
  });

  expect(contentCjsES2017).toMatchSnapshot('commonjs-es2017');
  expect(contentEsm).toMatchSnapshot('esm');
});

test('render fatherrc', async () => {
  const content = await renderService.render(ETplName.fatherrcTs, {});

  expect(content).toMatchSnapshot('fatherrc');
});

test('render package.json', async () => {
  const contentESM = await renderService.render(ETplName.packageJson, {
    name: 'project-name-0',
    author: 'Max <max@gmail.com>',
    isESM: true,
  });
  const contentDefault = await renderService.render(ETplName.packageJson, {
    name: 'project-name-1',
  });

  expect(contentESM).toMatchSnapshot('project0-esm-max');
  expect(contentDefault).toMatchSnapshot('project1-esm-max');
});
