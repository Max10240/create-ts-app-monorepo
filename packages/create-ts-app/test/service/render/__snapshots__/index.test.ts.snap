// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render fatherrc > fatherrc 1`] = `
"import { defineConfig } from \\"father\\";

export default defineConfig({
  platform: 'node',
  cjs: {},
  esm: {},
});
"
`;

exports[`render package.json > project0-esm-max 1`] = `
"{
  \\"name\\": \\"project-name-0\\",
  \\"type\\": \\"module\\",
  \\"version\\": \\"0.0.0\\",
  \\"description\\": \\"a project create by create-ts-app\\",
  \\"author\\": \\"Max <max@gmail.com>\\",
  \\"main\\": \\"dist/cjs/index.js\\",
  \\"module\\": \\"dist/esm/index.js\\",
  \\"types\\": \\"dist/esm/index.d.ts\\",
  \\"exports\\": {
    \\".\\": {
      \\"import\\": {
        \\"types\\": \\"./dist/esm/index.d.ts\\",
        \\"default\\": \\"./dist/esm/index.js\\"
      },
      \\"require\\": {
        \\"types\\": \\"./dist/esm/index.d.ts\\",
        \\"default\\": \\"./dist/cjs/index.js\\"
      }
    }
  },
  \\"sideEffects\\": false,
  \\"scripts\\": {
    \\"dev\\": \\"father dev\\",
    \\"build\\": \\"father build\\",
    \\"test\\": \\"vitest\\"
  },
  \\"files\\": [
    \\"dist\\"
  ]
}"
`;

exports[`render package.json > project1-esm-max 1`] = `
"{
  \\"name\\": \\"project-name-1\\",
  \\"version\\": \\"0.0.0\\",
  \\"description\\": \\"a project create by create-ts-app\\",
  \\"main\\": \\"dist/cjs/index.js\\",
  \\"module\\": \\"dist/esm/index.js\\",
  \\"types\\": \\"dist/esm/index.d.ts\\",
  \\"exports\\": {
    \\".\\": {
      \\"import\\": {
        \\"types\\": \\"./dist/esm/index.d.ts\\",
        \\"default\\": \\"./dist/esm/index.js\\"
      },
      \\"require\\": {
        \\"types\\": \\"./dist/esm/index.d.ts\\",
        \\"default\\": \\"./dist/cjs/index.js\\"
      }
    }
  },
  \\"sideEffects\\": false,
  \\"scripts\\": {
    \\"dev\\": \\"father dev\\",
    \\"build\\": \\"father build\\",
    \\"test\\": \\"vitest\\"
  },
  \\"files\\": [
    \\"dist\\"
  ]
}"
`;

exports[`render tsconfig > commonjs-es2017 1`] = `
"{
  \\"compilerOptions\\": {
    \\"target\\": target,
    \\"module\\": module,
    \\"moduleResolution\\": \\"nodenext\\",
    \\"moduleDetection\\": \\"force\\",
    \\"esModuleInterop\\": true,
    \\"strict\\": true,
    \\"skipLibCheck\\": true,
    \\"baseUrl\\": \\"./\\",
    \\"paths\\": {
      \\"@/*\\": [\\"./src/*\\"]
    },
    \\"rootDir\\": \\"src/\\",
    \\"noEmit\\": true,
    \\"declaration\\": true,
    \\"sourceMap\\": true
  }
}"
`;

exports[`render tsconfig > esm 1`] = `
"{
  \\"compilerOptions\\": {
    \\"target\\": target,
    \\"module\\": module,
    \\"moduleResolution\\": \\"nodenext\\",
    \\"moduleDetection\\": \\"force\\",
    \\"jsx\\": \\"react\\",
    \\"esModuleInterop\\": true,
    \\"strict\\": true,
    \\"skipLibCheck\\": true,
    \\"baseUrl\\": \\"./\\",
    \\"paths\\": {
      \\"@/*\\": [\\"./src/*\\"]
    },
    \\"rootDir\\": \\"src/\\",
    \\"noEmit\\": true,
    \\"declaration\\": true,
    \\"sourceMap\\": true
  }
}"
`;
